build-support/native-image/generate-native-image-for-pants-targets.bash

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env bash
+# Script to generate a platform-specific native-image of the pantsbuild zinc wrapper which works to
+# compile macros, from a pants project.
+
+# Requires `jq` and `go`. `jq` can be downloaded from https://stedolan.github.io/jq/!
+# Works on OSX and Linux -- see the README.md in this directory for more usage info.
+
+# TODO: This will be made automatic in pants via https://github.com/pantsbuild/pants/pull/6893.
+
+set -euxo pipefail
+
+# TODO: Right now, NATIVE_IMAGE_EXTRA_ARGS='-H:IncludeResourceBundles=org.scalactic.ScalacticBundle'
+# is necessary to build any zinc native-image which compiles any code using scalatest. This
+# information should be inferred by the native-image agent, but isn't yet.
+
+# TODO: build off of a more recent graal sha (more recent ones fail to build scalac and more): see
+# https://github.com/oracle/graal/issues/1448.
+
+
+# ARGUMENTS
+
+# $@: Forwarded to a `./pants compile` invocation which runs with reflection tracing enabled.
+# $NATIVE_IMAGE_EXTRA_ARGS: Forwarded to a `native-image` invocation.
+
+
+# CONSTANTS
+
+SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+# shellcheck source=build-support/native-image/build-zinc-native-image.bash
+source "${SCRIPT_DIR}/build-zinc-native-image.bash"
+
+GENERATED_CONFIG_DIR="$(normalize_path "${GENERATED_CONFIG_DIR:-generated-reflect-config}")"
+GENERATED_CONFIG_JSON_FILE="${GENERATED_CONFIG_DIR}/reflect-config.json"
+GENERATED_BUILD_FILE="${GENERATED_CONFIG_DIR}/BUILD"
+
+
+# FUNCTIONS
+
+function download_buildozer {
+  >&2 go get github.com/bazelbuild/buildtools/buildozer || return "$?"
+  which buildozer
+}
+
+function build_agent_dylib {
+  pushd "$(get_substratevm_dir)" >&2
+  if is_osx; then
+    desired_agent_dylib_path="$(pwd)/libnative-image-agent.dylib"
+  else
+    desired_agent_dylib_path="$(pwd)/libnative-image-agent.so"
+  fi
+  if [[ ! -f "$desired_agent_dylib_path" ]]; then
+    mx build >&2 || return "$?"
+    mx native-image --tool:native-image-agent --verbose >&2 || return "$?"
+    if is_osx; then
+      cp -v {,lib}native-image-agent.dylib || return "$?"
```

Comment:
Could pull out the extension as a variable to avoid needing the the second `if is_osx` (not important if you don't want to :))

Reply:
Will do!